cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/co/cask/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -329,6 +342,42 @@ public void runPipeline(PipelinePhase pipelinePhase, String sourcePluginType,
 
       emittedRecords.put(stageName, emittedBuilder.build());
     }
+
+    Collection<Future> sinkFutures = new ArrayList<>(sinkRunnables.size());
+    ExecutorService executorService = Executors.newFixedThreadPool(sinkRunnables.size(), new ThreadFactoryBuilder()
+      .setNameFormat("pipeline-sink-task")
+      .build());
+    for (Runnable runnable : sinkRunnables) {
+      sinkFutures.add(executorService.submit(runnable));
+    }
+
+    Throwable error = null;
+    Iterator<Future> futureIter = sinkFutures.iterator();
+    while (futureIter.hasNext()) {
+      Future future = futureIter.next();
+      try {
+        future.get();
+      } catch (ExecutionException | InterruptedException e) {
```

Comment:
If the exception is `InterruptedException`, even the task of the current `future` is not yet completed. The interruption is on the `get` call, not on the task itself.